framework/encode/custom_vulkan_api_call_encoders.cpp

```diff
@@ -652,24 +653,40 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateRayTracingPipelinesKHR(VkDevice
         omit_output_data = true;
     }
 
-    auto encoder = VulkanCaptureManager::Get()->BeginTrackedApiCallCapture(
-        format::ApiCallId::ApiCall_vkCreateRayTracingPipelinesKHR);
-    if (encoder)
-    {
-        encoder->EncodeHandleValue<DeviceWrapper>(device);
-        encoder->EncodeHandleValue<DeferredOperationKHRWrapper>(deferredOperation);
-        encoder->EncodeHandleValue<PipelineCacheWrapper>(pipelineCache);
-        encoder->EncodeUInt32Value(createInfoCount);
-        EncodeStructArray(encoder, pCreateInfos, createInfoCount);
-        EncodeStructPtr(encoder, pAllocator);
-        encoder->EncodeHandleArray<PipelineWrapper>(pPipelines, createInfoCount, omit_output_data);
-        encoder->EncodeEnumValue(result);
-        VulkanCaptureManager::Get()
-            ->EndGroupCreateApiCallCapture<VkDevice,
-                                           VkDeferredOperationKHR,
-                                           PipelineWrapper,
-                                           VkRayTracingPipelineCreateInfoKHR>(
-                result, device, deferredOperation, createInfoCount, pPipelines, pCreateInfos);
+    auto device_wrapper = GetWrapper<DeviceWrapper>(device);
+    if ((result != VK_OPERATION_DEFERRED_KHR) ||
+        (!device_wrapper->property_feature_info.feature_rayTracingPipelineShaderGroupHandleCaptureReplay))
+    {
+        // If the operation is not deferred by driver. or the system doesn't support
```

Comment:
As discussed, this block ends up being written twice by the code in this branch.  I don't think that's what we want for this solution.  If there has to be a deferred operation read of the output variable, we probably need a MetaData command to write that value, and then replay should handle the case of postponing creation of the VkPipeline handles until that MetaData command is written.

I'm confused about what's happening here.  Do you have a test case where the application creates an RT pipeline and gets VK_OPERATION_DEFERRED_KHR then joins the deferred operation until success, then uses the pipeline handles but GFXR somehow writes the operations out of order so the pipeline handles are used in replay before their creation is complete?

Reply:
> Do you have a test case where the application creates an RT pipeline and gets VK_OPERATION_DEFERRED_KHR then joins the deferred operation until success, then uses the pipeline handles but GFXR somehow writes the operations out of order so the pipeline handles are used in replay before their creation is complete?

If that's what's happening, then I think I need to understand how that could happen and we may need to introduce synchronization around the pipeline handles and anything using those pipeline handles, and not create this block twice.

Reply:
Can you describe in steps how the problem occurs so I can understand how this branch is expected to  the issue?

Reply:
The modified app is valid.

Yes, I think the hardware doesn't support the feature rayTracingPipelineShaderGroupHandleCaptureReplay, it triggers the issue that the above code block generates one more trace block. I'll add a commit for the issue. It may playback the trace of raytracingbasic as current dev branch. However, for this hardware, it's not guarantee playback always work, the reason is the following:

GFXR current framework requires the feature for capture/replay raytracing titles. It needs the related GPU address/ShaderGroupHandle capture/replay extension to reproduce these capture time GPU address/handle value during replay time. Without the feature (rayTracingPipelineShaderGroupHandleCaptureReplay) support, it's not guarantee for replaying raytracing trace file because shadergroup handle value may have different value during replay time, especially for complicated apps or real game, simple app and full trace file replay sometimes may work because the used GPU addresses/handles way less than real title.

Reply:
Done.